sentry/src/main/java/io/sentry/Scope.java

```diff
@@ -85,6 +90,11 @@ public final class Scope implements IScope {
 
   private @NotNull ISentryClient client = NoOpSentryClient.getInstance();
 
+  // TODO intended only for global scope
+  // TODO test for memory leak
```

Comment:
This needs testing once enough changes have been implemented to check for memory leaks.

Reply:
Agree, we might have to remove the entries that have been sent to `Sentry` from the map.

But this would have been an issue with `Hub` as well I think.